modules/internal/ChapelArray.chpl

```diff
@@ -1790,6 +1857,24 @@ module ChapelArray {
     }
   }  // record _array
 
+  //
+  // A helper function to check array equality (== on arrays promotes
+  // to an array of booleans)
+  //
+  proc _array.equals(that: _array) {
+    if this.rank != that.rank then
+      return false;
+    //    writeln("ranks match!");
+    for d in 1..this.rank do
+      if this.domain.dim(d).size != that.domain.dim(d).size then
+        return false;
+    //    writeln("sizes match");
+    //    writeln("this == that is: ", this == that);
+    return && reduce (this == that);
```

Comment:
Given that this is only being used on target locale arrays at present (order hundreds or thousands of elements, and not distributed), it seemed reasonable to use a reduction to me.  If/when someone were using this for a more significant computation, that might warrant rewriting it in an uglier way (or finally doubling down on reduction performance).